lib/semantics/resolve-names.cc

```diff
@@ -293,12 +294,12 @@ class ArraySpecVisitor {
 class ScopeHandler : public virtual ImplicitRulesVisitor {
 public:
   void set_rootScope(Scope &scope) { PushScope(scope); }
-  Scope &CurrScope() { return *scopes_.top(); }
+  Scope &currScope() { return *currScope_; }
```

Comment:
As we discussed, this member function isn't really an accessor.

Reply:
This still feels like a simple accessor to me. The data member should be a `Scope &` but the language doesn't allow that, so we have to work around it by using a never-null `Scope *`.